src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,10 +79,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+    UrlValidator defaultValidator = new UrlValidator();
+
+    String regex = "(https?://\\S+\\.(png|jpg|jpeg|gif|tiff|bmp)\\S*)";
+
+    System.out.println("Text is:" + userText);
+
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(userText);
+    String url = "";
+    while (matcher.find()) {
+      url = matcher.group();
+    }
+
+    if (defaultValidator.isValid(url)) {
+      String replacement = "<img src=\"$1\" />";
+      String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+      Message message = new Message(user, textWithImagesReplaced);
```

Comment:
Its good to keep code branches as short as possible. So for example, here,

```
Message message = new Message(...);
datastore.storeMessage(message);
```
is common between the two branches and could be simplified.

Reply:
Oh ok! So basically try to refactor out any code that is similar in the branches to avoid duplication, right?